src/gui/torrentcontentmodel.cpp

```diff
@@ -88,6 +99,42 @@ namespace
             return QIcon(iconPixmap);
         }
     };
+#elif defined(Q_OS_MAC)
+    // There is a similar bug on macOS, to be reported to Qt
+    // https://github.com/qbittorrent/qBittorrent/pull/6156#issuecomment-316302615
+    class MacFileIconProvider: public UnifiedFileIconProvider
+    {
+    public:
+        using QFileIconProvider::icon;
+        QIcon icon(const QFileInfo &info) const override
+        {
+            const auto suffix = info.suffix();
+            if (!suffix.isEmpty()) {
+                QPixmap pixmap = pixmapForExtension(suffix, QSize(32, 32));
```

Comment:
I am not an expert in underlying Qt implementation, but are you sure that won't break implicit move semantics at QIcon construction and thus affect the performance?..